### ISSUE TITLE
[cmake] for -DPLATFORM=DRM, unset CMP0054 causes (re)compilation failure

### ISSUE THREAD

**Issue by archie2x** [raysan5/raylib#2621]
- [X] I tested it on latest raylib version from master branch [b0f3a2c2]
 - [X] I checked there is no similar issue already reported
 - [X] My code has no errors or misuse of raylib

### Issue description

If PLATFORM=DRM and cmake files are edited and library (re)compiled, "PLATFORM" and dependent variables do not propagate correctly.

This only happens with DRM because "DRM" is itself a variable in the CMakeCache.txt and because of old, crufty, confusing reasons, documented in CMP0054.

The problem is that when you run cmake the 1st time, it creates an entry in CMakeCache.txt CALLED "DRM"
```
$ grep ^DRM build/linux/CMakeCache.txt 
DRM:FILEPATH=/usr/lib/x86_64-linux-gnu/libdrm.so
```

When run the 2nd time, Line 68 of cmake/LibraryConfigurations.cmake
```
elseif (${PLATFORM} MATCHES "DRM")
```
Expands 1st to:
```
elseif(DRM MATCHES DRM)
```
But, due to CMP0054 OLD behavior, the token "DRM" is further expanded to the value 
```
elseif(/usr/lib/x86_64-linux-gnu/libdrm.so MATCHES DRM)
```
Which it obviously doesn't. 

### Environment

I have reproduced this problem on two systems:

- Raspberry Pi 64bit, Ubuntu 20.04 Jammy, cmake 3.22.1, Ninja
- x86_64, Ubuntu 20.04 Focal, cmake 3.23.2, GNU Makefiles

### Issue Screenshot

See code example...

### Code Example

Configure build:

```
> cmake -B build/linux -DPLATFORM=DRM 
[...]
-- Audio Backend: miniaudio
-- Building raylib static library
-- Generated build type: Debug
-- Compiling with the flags:
--   PLATFORM=PLATFORM_DRM            ### <---- PLATFORM propagated
--   GRAPHICS=GRAPHICS_API_OPENGL_ES2
-- Building examples is enabled
[...]
```
And build:
```
> cmake --build build/linux
# build succeeds
```

Now edit any cmake file, this causes the next build to re-check all it's variables. Simulate here with 'touch.'

```
> touch cmake/CompileDefinitions.cmake # anything that causes cmake to re-run
> cmake --build build/linux
cmake --build build/linux/
-- Testing if -Werror=pointer-arith can be used -- compiles
-- Testing if -Werror=implicit-function-declaration can be used -- compiles
-- Testing if -fno-strict-aliasing can be used -- compiles
[...]
-- Audio Backend: miniaudio
-- Building raylib static library
-- Generated build type: Debug
-- Compiling with the flags:
--   PLATFORM=                                      ##### <===== PLATFORM IS MISSING
--   GRAPHICS=GRAPHICS_API_OPENGL_33
-- Building examples is enabled
-- Testing if -std=c11 can be used -- compiles
CMake Warning (dev) at examples/CMakeLists.txt:110 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "DRM" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
This warning is for project developers.  Use -Wno-dev to suppress it.

[...]
/usr/bin/cmake -S/Volumes/sensitive/raylib -B/Volumes/sensitive/raylib/build/linux --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /Volumes/sensitive/raylib/build/linux/CMakeFiles /Volumes/sensitive/raylib/build/linux//CMakeFiles/progress.marks
/usr/bin/make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/Volumes/sensitive/raylib/build/linux'
/usr/bin/make  -f raylib/CMakeFiles/raylib.dir/build.make raylib/CMakeFiles/raylib.dir/depend
make[2]: Entering directory '/Volumes/sensitive/raylib/build/linux'
cd /Volumes/sensitive/raylib/build/linux && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /Volumes/sensitive/raylib /Volumes/sensitive/raylib/src /Volumes/sensitive/raylib/build/linux /Volumes/sensitive/raylib/build/linux/raylib /Volumes/sensitive/raylib/build/linux/raylib/CMakeFiles/raylib.dir/DependInfo.cmake --color=
Dependencies file "raylib/CMakeFiles/raylib.dir/rcore.c.o.d" is newer than depends file "/Volumes/sensitive/raylib/build/linux/raylib/CMakeFiles/raylib.dir/compiler_depend.internal".
Consolidate compiler generated dependencies of target raylib
make[2]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
/usr/bin/make  -f raylib/CMakeFiles/raylib.dir/build.make raylib/CMakeFiles/raylib.dir/build
make[2]: Entering directory '/Volumes/sensitive/raylib/build/linux'
[  1%] Building C object raylib/CMakeFiles/raylib.dir/rcore.c.o
cd /Volumes/sensitive/raylib/build/linux/raylib && /usr/bin/cc -DGRAPHICS_API_OPENGL_33 -I/Volumes/sensitive/raylib/src -fno-strict-aliasing -Werror=implicit-function-declaration -Werror=pointer-arith  -g -std=gnu99 -MD -MT raylib/CMakeFiles/raylib.dir/rcore.c.o -MF CMakeFiles/raylib.dir/rcore.c.o.d -o CMakeFiles/raylib.dir/rcore.c.o -c /Volumes/sensitive/raylib/src/rcore.c
/Volumes/sensitive/raylib/src/rcore.c: In function ‘InitGraphicsDevice’:
/Volumes/sensitive/raylib/src/rcore.c:4612:22: error: ‘eglGetProcAddress’ undeclared (first use in this function)
 4612 |     rlLoadExtensions(eglGetProcAddress);
      |                      ^~~~~~~~~~~~~~~~~
/Volumes/sensitive/raylib/src/rcore.c:4612:22: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [raylib/CMakeFiles/raylib.dir/build.make:76: raylib/CMakeFiles/raylib.dir/rcore.c.o] Error 1
make[2]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
make[1]: *** [CMakeFiles/Makefile2:362: raylib/CMakeFiles/raylib.dir/all] Error 2
make[1]: Leaving directory '/Volumes/sensitive/raylib/build/linux'
make: *** [Makefile:166: all] Error 2
```

**Comment by archie2x** [comment]
Alternate fix is to set cmake_minimum_required to 3.1 instead of setting the policy explicitly

**Comment by raysan5** [comment]
@archie2x feel free to send a new PR if you think the second solution will be better iin the long term.
